Add a basic .editorconfig file

ac000 opened the pull request:
This just sets some basic file properties; character encoding, line endings, tabs vs spaces etc and is _not_ a replacement for a code formatter like indent(1) or clang-format.

avahahn commented on .editorconfig:
```diff
@@ -0,0 +1,17 @@
+root = true
+
+[{configure,{*.{c,cpp,h,go,java,js,py,rs}}]
```
is it worth including toml here?

ac000 replied:
If you think it's worth it, I'd put it under its own section like

```
[*.toml]
charset = utf-8
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true
```

As AFAICT it's just a flat INI style file with no indentation...

ac000 commented:
Add the`auto/` and `*.toml` files. The `auto` stuff uses a mixture of indentation styles (out of necessity) and the toml files are flat.

```
$ git range-diff 0287f011...2eec2fd0
1:  0287f011 ! 1:  2eec2fd0 Add a basic .editorconfig file
    @@ .editorconfig (new)
     +trim_trailing_whitespace = true
     +indent_style = tab
     +indent_size = 8
    ++
    ++[{auto/**,*.toml}]
    ++charset = utf-8
    ++end_of_line = lf
    ++insert_final_newline = true
    ++trim_trailing_whitespace = true
```

javorszky commented on .editorconfig:
```diff
@@ -0,0 +1,23 @@
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_style = tab
+indent_size = 8
```
Unusually wide, but otherwise not a blocker, merely an fyi

ac000 replied:
Unusual!?

Not if you use [Linux kernel coding style](https://docs.kernel.org/process/coding-style.html#codingstyle)...

> Tabs are 8 characters, and thus indentations are also 8 characters. There are heretic movements that try to make indentations 4 (or even 2!) characters deep, and that is akin to trying to define the value of PI to be 3.

javorszky replied:
Huh, today I learned. Thank you for the source and context :)

javorszky reviewed (approved):
I'm okay with this, and this would cut down on unnecessary whitespace changes

ac000 commented:
Add a section covering YAML

```
$ git range-diff 2eec2fd0...f68b6c07
1:  2eec2fd0 ! 1:  f68b6c07 Add a basic .editorconfig file
    @@ .editorconfig (new)
     +end_of_line = lf
     +insert_final_newline = true
     +trim_trailing_whitespace = true
    ++
    ++[*.yaml]
    ++charset = utf-8
    ++end_of_line = lf
    ++insert_final_newline = true
    ++trim_trailing_whitespace = true
    ++indent_style = space
    ++indent_size = 2
```

ac000 commented:
Rebase with master

```
$ git range-diff f68b6c07...06c4ea1f
 -:  -------- >  1:  57a75ea0 build(deps): bump openssl from 0.10.64 to 0.10.66 in /tools/unitctl
 -:  -------- >  2:  b892e994 tools/unitctl: update readme
 -:  -------- >  3:  1b484303 tools/unitctl: update readme
 -:  -------- >  4:  e743b6ce tools/unitctl: remove (default) from option text
 -:  -------- >  5:  2a243740 tools/unitctl: make json-pretty default output fmt
 -:  -------- >  6:  43faf99d tools/unitctl: reword freeform message for output
 -:  -------- >  7:  a91b961d tools/unitctl: make application directory configurable
 -:  -------- >  8:  e56c4ede fuzzing: code cleanup
 -:  -------- >  9:  900d25c3 fuzzing: fixed harness bug
 -:  -------- > 10:  bc49274d fuzzing: updated JSON target
 -:  -------- > 11:  3667c3e2 fuzzing: added new basic targets
 1:  f68b6c07 = 12:  06c4ea1f Add a basic .editorconfig file
```
